src/store/datasets-files.ts

```diff
@@ -1,8 +1,55 @@
 import { del, set } from '@vue/composition-api';
 import { defineStore } from 'pinia';
+import { pluck } from '../utils';
+
+export type DatasetPath = string & { __type: 'DatasetPath' };
+export type DatasetUrl = string & { __type: 'DatasetUrl' };
+export type LocalDatasetFile = { file: File };
+export type ZipDatasetFile = LocalDatasetFile & { path: string };
```

Comment:
I think we should pick a different name other than "path". I think it's a bit vague. Maybe `archivePath`? We will need to rewrite its usage (I think it's used in serialization/deserialization).